src/modules/firewall/src/lib/Firewall.h

```diff
@@ -39,32 +40,167 @@ class FirewallLog
     static OSCONFIG_LOG_HANDLE m_logHandle;
 };
 
+class GenericRule
+{
+public:
+    enum class State
+    {
+        Present,
+        Absent
+    };
+
+    enum class Action
+    {
+        Accept,
+        Drop,
+        Reject
+    };
+
+    enum class Direction
+    {
+        In,
+        Out
+    };
+
+    enum class Protocol
+    {
+        Any,
+        Tcp,
+        Udp,
+        Icmp,
+    };
```

Comment:
If we proceed with enum of strings in MIM, why not use enum of strings in implementation as well?